plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -872,37 +896,44 @@ void TransformationMatrix::initialiseToPrim(bool readFromPrim, Transform* transf
     m_flags |= kInheritsTransform;
 
   if(matchesMayaProfile(m_xformops.begin(), m_xformops.end(), m_orderedOps.begin()))
-  {
     m_flags |= kFromMayaSchema;
-  }
-  else
-  {
-  }
+
+  // Disable push to prim if enabled, otherwise MPlug value queries and setting
+  // in the switch statement below will trigger pushing to the prim, which
+  // will create undesirable usd "over"s.
+  //
+  bool pushOriginallyEnabled = pushToPrimEnabled();
+  m_flags &= ~kPushToPrimEnabled;
 
```

Comment:
Could you explain what a RAll object is?

Reply:
In this case It would mean creating an object that does the cleanup when it goes out of scope - just by having the same cleanup code as below in the destructor. This would be triggered when the code block complete (i.e normal execution) but also cover the case when an exception is thrown

Reply:
Yup, what Eoin said - more details here: https://en.cppreference.com/w/cpp/language/raii
The acronym is pretty opaque, but basically because c++ doesn’t have a try / finally statement, it’s a trick to use the deconstructor to serve the same purpose, some those are supposed to run even when unwinding the stack for exceptions.

Reply:
Yup, what Eoin said - more details here: https://en.cppreference.com/w/cpp/language/raii
The acronym is pretty opaque, but basically because c++ doesn’t have a try / finally statement, it’s a trick to use the deconstructor to serve the same purpose, some those are supposed to run even when unwinding the stack for exceptions.

Reply:
Instead of complicating things with an extra object, I just used a try/catch block where the catch clause resets the original state and rethrows the exception in case something else is looking for it.  Hopefully that's acceptable.